Lib/Common/Table/TableEntity.cs

```diff
@@ -108,19 +108,26 @@ public TableEntity(string partitionKey, string rowKey)
         /// <param name="operationContext">An <see cref="OperationContext"/> object that represents the context for the current operation.</param>
         public virtual void ReadEntity(IDictionary<string, EntityProperty> properties, OperationContext operationContext)
```

Comment:
Previously, this method contained all of the logic to read an entity. It seemed already prepared to handle objects of arbitrary types, but "this" was always passed into the reflection-y methods, effectively forcing it to always work against the _TableEntity_ type (or whatever subclass the user had created).

I simply pulled this code out into another method that allows you to pass in the object you want to use in place of "this" and called that method from here with "this" to keep the existing functionality intact.